cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -0,0 +1,389 @@
+# Introduction
+
+The objective of the cheat sheet is to provide advices regarding the protection against [Server Side Request Forgery](https://www.acunetix.com/blog/articles/server-side-request-forgery-vulnerability/) (SSRF) attack.
+
+This cheat sheet will focus on the defensive point of view and will not explain how to perform this attack. This [talk](../assets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet_Orange_Tsai_Talk.pdf) from the security researcher [Orange Tsai](https://twitter.com/orange_8361) as well as this [document](../assets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet_SSRF_Bible.pdf) provide techniques on how to perform this kind of attack.
+
+# Context
+
+SSRF is an attack vector that abuses an application to interact with the internal/external network or the machine itself. One of the enablers for this vector is the mishandling of URLs, as showcased in the following examples: 
+- Image on external server (*e.g.* user enters image URL of their avatar for the application to download and use).
+- Custom [WebHook](https://en.wikipedia.org/wiki/Webhook) (user have to specify WebHook handlers or Callback URLs).
+- Request to another application, often located on another network, to perform a specific task. Depending of the business case, it can happen that information from the user are needed to perform the action.
+
+## Overview of an SSRF common flow
+
+![SSRFCommonFlow](../assets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet_SSRF_Common_Flow.png)
+
+*Note:* SSRF is not limited to the HTTP protocol, despite the fact that in general the first request leverages it, yet the second request is performed by the application itself, and thus it could be using different protocols (*e.g.* FTP, SMB, SMTP, etc.). The protocol usage is highly dependent on the application’s requirements.
+
+# Cases
+
+Depending on the application's functionality and requirements, there are two basic cases in which SSRF can happen:
+* Application can send request only to **identified and trusted applications**: Case when [whitelist](https://en.wikipedia.org/wiki/Whitelisting) approach is available.
+* Application can send requests to **ANY other IP address or domain name**: Case when [whitelist](https://en.wikipedia.org/wiki/Whitelisting) approach is not available.
```

Comment:
To be more straight this case is about 'Application can send requests to ANY **external** IP address or domain name.

Reply:
Change applied and a new commit has been pushed.